Extension/src/main.ts

```diff
@@ -11,330 +11,135 @@ import * as util from './common';
-    // and also show the C/C++ output pane and the "wait for download" message.
-    tempCommands = [];
-    delayedCommandsToExecute = new Set<string>();
-    registerTempCommand("C_Cpp.ConfigurationEdit");
-    registerTempCommand("C_Cpp.ConfigurationSelect");
-    registerTempCommand("C_Cpp.SwitchHeaderSource");
-    registerTempCommand("C_Cpp.Navigate");
-    registerTempCommand("C_Cpp.GoToDeclaration");
-    registerTempCommand("C_Cpp.PeekDeclaration");
-    registerTempCommand("C_Cpp.ToggleErrorSquiggles");
-    registerTempCommand("C_Cpp.ToggleIncludeFallback");
-    registerTempCommand("C_Cpp.ShowReleaseNotes");
-    registerTempCommand("C_Cpp.ResetDatabase");
-    registerTempCommand("C_Cpp.PauseParsing");
-    registerTempCommand("C_Cpp.ResumeParsing");
-    registerTempCommand("C_Cpp.ShowParsingCommands");
-    registerTempCommand("C_Cpp.TakeSurvey");
-
-    processRuntimeDependencies(() => {
-        downloadCpptoolsJsonPkg().then(() => {
-            util.readFileText(util.getExtensionFilePath("cpptools.json"))
-                .then((cpptoolsString) => {
-                    processCpptoolsJson(cpptoolsString);
-                })
-                .catch((error) => {
-                    // We already log telemetry if cpptools.json fails to download.
-                })
-                .then(() => {
-                    // Main activation code.
-                    tempCommands.forEach((command) => {
-                        command.dispose();
-                    });
-                    tempCommands = [];
-                    LanguageServer.activate(delayedCommandsToExecute);
-                    delayedCommandsToExecute.forEach((command) => {
-                        vscode.commands.executeCommand(command);
-                    });
-                    delayedCommandsToExecute.clear();
-                });
-        });
-    });
-
-    setInterval(() => {
-        // Redownload occasionally to prevent an extra reload during long sessions.
-        downloadCpptoolsJsonPkg();
-    }, 30 * 60 * 1000); // 30 minutes.
+    return processRuntimeDependencies();
 }
 
 export function deactivate(): Thenable<void> {
     DebuggerExtension.dispose();
-
-    tempCommands.forEach((command) => {
-        command.dispose();
-    });
-
     Telemetry.deactivate();
     return LanguageServer.deactivate();
 }
 
-function removePotentialPII(str: string): string {
-    let words: string[] = str.split(" ");
-    let result: string = "";
-    for (let word of words) {
-        if (word.indexOf(".") == -1 && word.indexOf("/") == -1 && word.indexOf("\\") == -1 && word.indexOf(":") == -1) {
-            result += word + " ";
-        } else {
-            result += "? ";
+async function processRuntimeDependencies(): Promise<void> {
+    const installLockExists: boolean = await util.checkInstallLockFile();
+
+    try {
+        const info: PlatformInformation = await PlatformInformation.GetPlatformInformation();
```

Comment:
GetPlatformInformation doesn't need to be called when we aren't installing the extension, so perhaps this call should be made independently in `offlineInstallation` and `onlineInstallation`.